src/test/java/org/dataloader/DataLoaderMappedBatchPublisherTest.java

```diff
@@ -0,0 +1,175 @@
+package org.dataloader;
+
+import org.junit.Test;
+import reactor.core.publisher.Flux;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.awaitility.Awaitility.await;
+import static org.dataloader.DataLoaderFactory.newMappedPublisherDataLoader;
+import static org.dataloader.DataLoaderOptions.newOptions;
+import static org.dataloader.fixtures.TestKit.listFrom;
+import static org.dataloader.impl.CompletableFutureKit.cause;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class DataLoaderMappedBatchPublisherTest {
+
+    MappedBatchPublisher<String, String> evensOnlyMappedBatchLoader = (keys, subscriber) -> {
+        Map<String, String> mapOfResults = new HashMap<>();
+
+        AtomicInteger index = new AtomicInteger();
+        keys.forEach(k -> {
+            int i = index.getAndIncrement();
+            if (i % 2 == 0) {
+                mapOfResults.put(k, k);
+            }
+        });
+        Flux.fromIterable(mapOfResults.entrySet()).subscribe(subscriber);
+    };
+
+    private static <K, V> DataLoader<K, V> idMapLoader(DataLoaderOptions options, List<Collection<K>> loadCalls) {
+        MappedBatchPublisher<K, V> kvBatchLoader = (keys, subscriber) -> {
+            loadCalls.add(new ArrayList<>(keys));
+            Map<K, V> map = new HashMap<>();
+            //noinspection unchecked
+            keys.forEach(k -> map.put(k, (V) k));
+            Flux.fromIterable(map.entrySet()).subscribe(subscriber);
+        };
+        return DataLoaderFactory.newMappedPublisherDataLoader(kvBatchLoader, options);
+    }
+
+    private static <K, V> DataLoader<K, V> idMapLoaderBlowsUps(
+        DataLoaderOptions options, List<Collection<K>> loadCalls) {
+        return newMappedPublisherDataLoader((MappedBatchPublisher<K, V>) (keys, subscriber) -> {
+            loadCalls.add(new ArrayList<>(keys));
+            Flux.<Map.Entry<K, V>>error(new IllegalStateException("Error")).subscribe(subscriber);
+        }, options);
```

Comment:
We need a test for the case where say 1/2 the values are sent AND then it blows up.  So that we error out ONLY those that have not completed.

ps.  you may have this but I am wrting this comment as I think of the use case.